aten/src/ATen/native/sparse/cuda/SparseBlas.cpp

```diff
@@ -57,6 +57,7 @@ Tensor& sparse_sampled_addmm_out_sparse_csr_cuda(
 
   // there's a segfault when calling cuSPARSE on 0-sized matrices
   if (mat1.numel() == 0 || mat2.numel() == 0) {
+    result.mul_(beta);
```

Comment:
The exit condition needs to be a bit more nuanced. Sorry I didn't catch this earlier. 


From the docs: 
>If beta is 0, then input will be ignored, and nan and inf in it will not be propagated.

```suggestion
    // According to docs, when beta==0 values in self should be ignored.
    // nans and infs should not propagate
    if(beta.toComplexDouble() == 0.){
        result.zero_();
    } else {
        result.mul_(beta);
    }
```